app/controllers/products_controller.rb

```diff
@@ -0,0 +1,71 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: %i[show destroy update edit]
+  before_action :set_associations, only: %i[new edit]
+
+  def index
+    @products = Product.includes(:supplier, :category).order(:id).all
+  end
+
+  def show; end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new
+    @product.assign_attributes(permitted_attributes)
+    @product.assign_attributes(units_on_order: 0)
```

Comment:
Skoro to nadpisujemy to dlaczego w ogóle dopuszczamy ten parametr w permitted attributes? Nie widzę aby go można było wysłać także przez formularz. Czy to znaczy ile ogólnie zamówiono jednostek danego produktu? Jeśli tak to powinniśmy usunąć ten atrybut z `permitted_attributes`. Przydałoby się też zwiększać tą wartość z każdym złożonym zamówieniem na dany produkt. Ale to można zrobić już potem. :wink:

Reply:
Usunąłem `units_on_order` - przeoczenie.
Istotnie, przy składaniu zamówienia należałoby zmieniać wartość tego parametru, ale tu by trzeba się zastanowić nad logiką - wydaje mi się, że powinien on odzwierciedlać stan, kiedy produkt nie jest jeszcze wysłany (jest w magazynie), ale już jest przypisany do zamówienia. Myślę, że na tym etapie nie będziemy tego zmieniać.